zipkin-receiver-kafka/src/main/scala/com/twitter/zipkin/receiver/kafka/SpanDecoder.scala

```diff
@@ -0,0 +1,22 @@
+package com.twitter.zipkin.receiver.kafka
+
+import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
+import com.twitter.scrooge.BinaryThriftStructSerializer
+import com.twitter.zipkin.conversions.thrift.{thriftSpanToSpan, spanToThriftSpan}
+import kafka.message.Message
+
+class SpanDecoder extends KafkaProcessor.KafkaDecoder {
+  val deserializer = new BinaryThriftStructSerializer[ThriftSpan] {
+    def codec = ThriftSpan
+  }
+
+  def toEvent(message: Message): Option[List[ThriftSpan]] = {
+
+    val buffer = message.payload
+    val payload = new Array[Byte](buffer.remaining)
+    buffer.get(payload)
+    val span = deserializer.fromBytes(payload)
+    Some(List(span))
```

Comment:
see here? you are in charge of creating `List()`. if you were afraid of getting a null, this would have to be `Option()` not `Some()`, since `Some(null)` gives Some(null) (vs `Option(null)` which gives `None`).

```
scala> Option(null)
res2: Option[Null] = None

scala> Some(null)
res3: Some[Null] = Some(null)
```